### ISSUE TITLE
Let wpb.sh update $PATH

### ISSUE THREAD

**Issue by mecab** [greymd/ttcopy#11]
As discussed  as #8, `wpb.sh` should put their directory into `$PATH`, so that the users can use the functions from other programs.

**Comment by greymd** [comment]
There is the `zplug` setting for that.
https://github.com/zplug/zplug/wiki/Configurations

```sh
# Specify relative directory to add to $PATH
zplug "rcmdnk/sentaku", as:command, use:"bin/*"
```

**Comment by mecab** [comment]
That's nice!

How can we provide easy way to use for non-zplug users? Providing `ttcp_activate.sh` to add the directory into the PATH?

**Comment by greymd** [comment]
> How can we provide easy way to use for non-zplug users?

yeah, that's the problem.
こことても悩みどころ。どっちがいいですかね。。。
I have two ideas.

## 1. Edit `PATH` directly.

```
.
├── bin
│   ├── ttcopy
│   └── ttpaste
├── ttcp.sh
├── ttcopy.plugin.zsh
└── ttcp_activate.sh
```

And `ttcp_activate.sh` is like this.

```bash
#!/bin/bash
_TTCP_DIR="$(cd "$(dirname "${BASH_SOURCE[0]:-${(%):-%N}}")"; pwd)"
export PATH="$_TTCP_DIR/bin:$PATH"
```

#### pros
* Easy to import. Just add this line to `.bashrc`.

```
source $path_to_repository/greymd/ttcp_activate.sh
```

#### cons
* Users cannot change the location of git repository easily.
  + Users may think `ttcopy` is executable by itself. Then, they may move commands under `/usr/local/bin`. But it does not work..

## 2. Prepare builder
発想が古いけど、`ttcopy`と`ttpaste`を単体で実行可能にビルドして`PATH`に置いちゃう方法もありかなと思ったり。
Prepare `Makefile`.
If users execute `make install`, our script inserts `ttcp.sh` into `bin/ttcopy` and `bin/ttpaste` and put them under `/usr/local/bin`. Then, `ttcopy` and `ttpaste` will be executable by themselves.



```
.
├── Makefile
├── bin
│   ├── ttcopy
│   └── ttpaste
├── ttcp.sh
└── ttcopy.plugin.zsh
```

#### pros
* `.bashrc` does not have any dependencies. Even a user changes the location of the git repository, there is no impacts.

#### cons
* Lazy to install & uninstall.
* Lazy to update the version.

**Comment by mecab** [comment]
How about

```
.
├── bin
│   ├── ttcopy
│   └── ttpaste
├── lib
│   └── ttcp.sh
├── ttcopy.plugin.zsh
└── ttcp_activate.sh
```

with adding some modification to `_TTCP_DIR`
so then the users can copy to /usr/local/* directly.

Another idea is use 

```bash
_TTCP_DIR="$(cd "$(dirname "$(realpath ${BASH_SOURCE[0]:-${(%):-%N}})")"; pwd)"
```

And tell user to use symlink instead of copy to put into /usr/local/bin. Above code resolve the files actual path even if it was symlink.
(Then we might have to consider about POSIX again 👮)

**Comment by mecab** [comment]
By f2f conversation, we agreed on `/bin` and `/lib` tree, but not using `realpath` to support symlink.

**Comment by greymd** [comment]
Closed by #33
